src/components/CodeView/index.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+
+import styles from './styles.module.scss';
+import { getLanguage, getLines, mapWithDepth } from './utils';
+import refractor from '../../refractor';
+
+type PublicProps = {
+  mimeType: string;
+  content: string;
+};
+
+// This function mimics what https://github.com/rexxars/react-refractor does,
+// but we need a different layout to inline comments so we cannot use this
+// component.
+const renderHighlightedCode = (code: string, language: string) => {
+  const ast = refractor.highlight(code, language);
+  const value = ast.length === 0 ? code : ast.map(mapWithDepth(0));
+
+  return (
+    <pre className={styles.highlightedCode}>
+      <code className={`language-${language}`}>{value}</code>
+    </pre>
+  );
+};
+
+const CodeViewBase = ({ mimeType, content }: PublicProps) => {
+  const language = getLanguage(mimeType);
+
+  return (
+    <div className={styles.CodeView}>
+      <table className={styles.table}>
```

Comment:
Why did you use a table? To match `react-diff-view`? A css grid would probably be easier to deal with.

Reply:
Since we render data, `table` does make sense. That's what GitHub uses too (and yes, react-diff view also uses table).

Reply:
Well, I imagine github only uses tables for historic reasons and maybe to support IE. I guess syncing with `react-diff` is a good goal but I find that CSS grid is more flexible for these types of layouts.

Reply:
I can file an issue to switch to a grid but I really think it is simpler as a table, and it works.

We will have to inline fullwidth rows for the comments, and doing this with a grid, when you don't know the number of rows in the css, seems though. I initially tried the grid approach, then I looked at github and react-diff-view and convinced myself that table was the right approach.